crypto/fipsmodule/service_indicator/service_indicator_test.cc

```diff
@@ -131,17 +164,15 @@ TEST(ServiceIndicatorTest, AESGCM) {
   ASSERT_TRUE(EVP_AEAD_CTX_init(aead_ctx.get(), EVP_aead_aes_128_gcm_randnonce(),
                                 kAESKey, sizeof(kAESKey), 0, nullptr));
 
-  // AES-GCM Encryption
   CALL_SERVICE_AND_CHECK_APPROVED(approved,EVP_AEAD_CTX_seal(aead_ctx.get(),
       encrypt_output, &out_len, sizeof(encrypt_output), nullptr, 0, kPlaintext, sizeof(kPlaintext), nullptr, 0));
-  ASSERT_TRUE(approved);
-  
-  // AES-GCM Decryption
+  ASSERT_EQ(approved, AWSLC_APPROVED);
+
   CALL_SERVICE_AND_CHECK_APPROVED(approved,EVP_AEAD_CTX_open(aead_ctx.get(),
       decrypt_output, &out2_len, sizeof(decrypt_output), nullptr, 0, encrypt_output, out_len, nullptr, 0));
   ASSERT_TRUE(check_test(kPlaintext, decrypt_output, sizeof(kPlaintext),
                   "AES-GCM Decryption for Internal IVs"));
-  ASSERT_TRUE(approved);
+  ASSERT_EQ(approved, AWSLC_APPROVED);
 }
 
 #else
```

Comment:
I suggest you change the comment of the following line (feel free to reword it): 
"Service indicator calls should not be used in non-FIPS builds. However, in that case, if the macro |CALL_SERVICE_AND_CHECK_APPROVED| is used, it will return AWSLC_APPROVED, but the direct calls to |FIPS_service_indicator_xxx| will not indicate approved state."